### PR TITLE
dotCMS/core#16767 Set edit mode and save content to selected persona

### PR DESCRIPTION
**Pull request by freddyucv**

### CODE REVIEW

File: src/app/portlets/dot-edit-page/content/components/dot-edit-page-view-as-controller/dot-edit-page-view-as-controller.component.ts [hunk 1]
```diff
@@ -46,6 +46,10 @@ export class DotEditPageViewAsControllerComponent implements OnInit {
             .subscribe((messages: { [key: string]: string }) => {
                 this.messages = messages;
             });
+
+        this.dotPageStateService.contentAdded$.subscribe(() => {
+            this.pageState.state.haveContent = true;
```
Comment (fmontes) [inline src/app/portlets/dot-edit-page/content/components/dot-edit-page-view-as-controller/dot-edit-page-view-as-controller.component.ts]: Can we have side effect for changing the state of the page like this? Do we need a take one? what happen when we add a content and then remove it?

File: src/app/portlets/dot-edit-page/content/services/dot-page-state/dot-page-state.service.ts [hunk 2]
```diff
@@ -22,6 +22,7 @@ import { DotRouterService } from '@services/dot-router/dot-router.service';
 @Injectable()
 export class DotPageStateService {
     state$: Subject<DotRenderedPageState> = new Subject<DotRenderedPageState>();
+    contentAdded$ = new Subject<DotRenderedPageState>();
```
Comment (fmontes) [inline src/app/portlets/dot-edit-page/content/services/dot-page-state/dot-page-state.service.ts]: If you already have a place where you can tell if content were added or not, why do you need another subject here?

File: src/app/view/components/dot-persona-selected-item/dot-persona-selected-item.component.html [hunk 3]
```diff
@@ -1,4 +1,8 @@
-<div class="dot-persona-selector__container">
+<div
+    class="dot-persona-selector__container"
+    [pTooltip]="disabled ? 'Add content to start personalizing' : null"
```
Comment (fmontes) [inline src/app/view/components/dot-persona-selected-item/dot-persona-selected-item.component.html]: Needs translation.

File: src/app/portlets/dot-edit-page/shared/models/dot-rendered-page-state.model.ts [hunk 4]
```diff
@@ -63,7 +64,13 @@ export class DotRenderedPageState {
         return this._user;
     }
 
-    set dotRenderedPageState(dotRenderedPageState: DotRenderedPageState) {
+    set dotRenderedPageState(dotRenderedPageState: DotPageRender) {
         this.dotRenderedPage = dotRenderedPageState;
     }
+
+    get dotRenderedPageState() {
```
Comment (fmontes) [inline src/app/portlets/dot-edit-page/shared/models/dot-rendered-page-state.model.ts]: But, but, but... why?

File: src/app/portlets/dot-edit-page/content/services/dot-page-state/dot-page-state.service.ts [hunk 5]
```diff
@@ -198,6 +199,10 @@ export class DotPageStateService {
         }
     }
 
+    contentAdded(): void {
```
Comment (alfredo-dotcms) [inline src/app/portlets/dot-edit-page/content/services/dot-page-state/dot-page-state.service.ts]: doc?

File: src/app/portlets/dot-edit-page/shared/models/dot-rendered-page.model.ts [hunk 6]
```diff
@@ -3,11 +3,48 @@ import { DotPage } from './dot-page.model';
 import { DotTemplate } from './dot-template.model';
 import { DotEditPageViewAs } from '@models/dot-edit-page-view-as/dot-edit-page-view-as.model';
 
-export interface DotPageRender {
+export interface DotPageRenderResponse {
```
Comment (freddyucv) [inline src/app/portlets/dot-edit-page/shared/models/dot-rendered-page.model.ts]: I need a better name for this interface

File: src/app/portlets/dot-edit-page/content/components/dot-edit-page-view-as-controller/dot-edit-page-view-as-controller.component.html [hunk 7]
```diff
@@ -22,6 +22,7 @@
         ></dot-device-selector>
         <dot-persona-selector
             (delete)="deletePersonalization($event)"
+            [disabled]="!(dotPageStateService.hasContent$ | async)"
```
Comment (fmontes) [inline src/app/portlets/dot-edit-page/content/components/dot-edit-page-view-as-controller/dot-edit-page-view-as-controller.component.html]:
```suggestion
            [disabled]="!(dotPageStateService.haveContent$ | async)"
```

File: src/app/portlets/dot-edit-page/shared/models/dot-rendered-page.model.ts [hunk 8]
```diff
@@ -3,11 +3,48 @@ import { DotPage } from './dot-page.model';
...
+
+    get canCreateTemplate(): boolean {
+        return this.params.canCreateTemplate;
+    }
+
+    get viewAs(): DotEditPageViewAs {
+        return this.params.viewAs;
+    }
+
+    get numberContents(): number {
```
Comment (fmontes) [inline src/app/portlets/dot-edit-page/shared/models/dot-rendered-page.model.ts]: This shouldn't be here, this is part of the state, this class is just for wrap the response and that we can extend the state, but this should be exactly as the response.

File: src/app/portlets/dot-edit-page/content/dot-edit-content.component.ts [hunk 9]
```diff
@@ -468,6 +468,15 @@ export class DotEditContentComponent implements OnInit, OnDestroy {
             });
     }
 
+    private callPageStateServiceMethod(event: PageModelChangeEvent) {
+        console.log('event.type', event.type);
+        if (event.type === PageModelChangeEventType.ADD_CONTENT) {
+            this.dotPageStateService.contentAdded();
```
Comment (fmontes) [inline src/app/portlets/dot-edit-page/content/dot-edit-content.component.ts]: The service could have this method inside, so we can pass the event there and he can add or remove to update the have content. Is cleaner.

File: src/app/portlets/dot-edit-page/shared/models/dot-rendered-page.model.ts (hunk 6 above)
Comment (fmontes) [reply src/app/portlets/dot-edit-page/shared/models/dot-rendered-page.model.ts]: 😑

**Pull request merged**